src/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -981,10 +983,24 @@ public static unsafe object CheckCast(void* pvTargetType, object obj)
                 return CheckCastArray(pvTargetType, obj);
             else if (pTargetType->IsInterface)
                 return CheckCastInterface(pvTargetType, obj);
+            else if (pTargetType->IsParameterizedType)
```

Comment:
This logic is duplicated in multiple other places, e.g. `RhGetRuntimeHelperForType`

Reply:
None of those seem to matter without universal shared code, but I guess it doesn't hurt fixing.